### PR TITLE
Standardize ticket enums and edge types

### PR DESCRIPTION
**Pull request by darshangodase**
## Description

[Provide a brief description of the changes in this PR]

### Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update (Is there an addition/change in API Request, Response? If yes, documentation is required)
- [ ] Performance improvement
- [ ] Code refactoring
- [ ] Security fix

## Related Issues

- Fixes #[issue number]
- Closes #[issue number]
- Related to #[issue number]

## How Has This Been Tested?

[Describe the tests you ran to verify your changes. Provide instructions so we can reproduce.]

### Test Configuration

- [ ] Unit tests pass
- [ ] Integration tests pass
- [ ] Manual testing completed
- [ ] Cross-browser testing (if applicable)
- [ ] Mobile responsiveness tested (if applicable)

## Core Functionality Testing

Please confirm that the following core functionalities are working as expected:

- [ ] **Search capabilities** are working correctly
- [ ] **Knowledge search** is functioning properly
- [ ] **Connector indexing** is working as expected
- [ ] **Citations** are displaying and linking correctly
- [ ] **Documentation** is updated at https://docs.pipeshub.com/introduction

## What You Have Tested

Please describe what you have specifically tested:

- [ ] Feature works in development environment
- [ ] Feature works in staging environment
- [ ] Error handling scenarios tested
- [ ] Edge cases considered and tested
- [ ] Performance impact assessed
- [ ] Security implications reviewed

### Test Results

[Provide details of your test results here]

## Screenshots/Videos

**Before:**
[Attach screenshots/videos showing the current behavior]

**After:**
[Attach screenshots/videos showing the new behavior]

**Note:** Screenshots are required before merge for UI changes.

## Code Quality Checklist

- [ ] My code follows the project's coding standards
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

## Documentation

- [ ] Documentation has been updated (if applicable)
- [ ] API documentation updated (if applicable)
- [ ] README updated (if applicable)
- [ ] Changelog updated (if applicable)

## Security Considerations

- [ ] No sensitive data is exposed
- [ ] Input validation is implemented where necessary
- [ ] Authentication/authorization is properly handled
- [ ] No security vulnerabilities introduced

## Breaking Changes

- [ ] This PR introduces breaking changes
- [ ] Migration guide provided (if applicable)
- [ ] Version bump required

If breaking changes are introduced, please describe them here:

[Describe any breaking changes and how users should adapt]

## Performance Impact

- [ ] No performance impact
- [ ] Performance improved
- [ ] Performance impact assessed and acceptable

[Describe any performance implications]

## Dependencies

- [ ] No new dependencies added
- [ ] New dependencies are necessary and approved
- [ ] Dependencies updated and tested

List any new dependencies:
- [Dependency name and version]

## Deployment Notes

[Any special deployment considerations or steps]

## Checklist Before Merge

- [ ] All tests are passing
- [ ] Code review completed and approved
- [ ] Documentation updated and reviewed
- [ ] Screenshots/videos attached for UI changes
- [ ] Core functionality verified
- [ ] Security review completed (if applicable)
- [ ] Performance impact assessed
- [ ] Ready for production deployment

## Additional Notes

[Any additional information that would be helpful for reviewers]

---

**For Reviewers:**

Please ensure all checklist items are completed before approving this PR. Pay special attention to:
- Core functionality testing
- Security implications
- Performance impact
- Documentation completeness

### CODE REVIEW

File: backend/python/app/connectors/utils/ticket_mapper.py [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+def map_ticket_status(api_status: Optional[str], custom_mappings: Optional[Dict[str, TicketStatus]] = None) -> Optional[TicketStatus]:
+    """Quick function to map status without creating a mapper instance"""
+    mapper = TicketValueMapper(status_mappings=custom_mappings)
+    return mapper.map_status(api_status)
+
+
+def map_ticket_priority(api_priority: Optional[str], custom_mappings: Optional[Dict[str, TicketPriority]] = None) -> Optional[TicketPriority]:
+    """Quick function to map priority without creating a mapper instance"""
+    mapper = TicketValueMapper(priority_mappings=custom_mappings)
+    return mapper.map_priority(api_priority)
```
[the remark this answers is not in the document]
Comment (darshangodase) [reply backend/python/app/connectors/utils/ticket_mapper.py]: fiixed

File: backend/python/app/connectors/utils/ticket_mapper.py [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+            elif num == 2:
+                return TicketPriority.HIGH
+            elif num == 3:
+                return TicketPriority.MEDIUM
+            elif num == 4:
+                return TicketPriority.LOW
+            elif num >= 5:
+                return TicketPriority.LOWEST
+        except ValueError:
+            pass
```
[the remark this answers is not in the document]
Comment (darshangodase) [reply backend/python/app/connectors/utils/ticket_mapper.py]: fixed

File: backend/python/app/config/constants/arangodb.py [hunk 3]
```diff
@@ -261,6 +262,33 @@ class RecordRelations(Enum):
...
+    SPLIT_FROM = "SPLIT_FROM"  # This ticket was split from the linked ticket
+    MERGED_INTO = "MERGED_INTO"  # This ticket was merged into the linked ticket
+    UNKNOWN = "UNKNOWN"  # Unknown or unmapped relationship type
+
+
+class TicketEdgeTypes(Enum):
+    """Standard edge types for ticket relationships"""
+    ASSIGNED_TO = "ASSIGNED_TO"  # Ticket -> User (assignee)
+    REPORTED_BY = "REPORTED_BY"  # Ticket -> User (reporter)
+    CREATED = "CREATED_BY"  # Ticket -> User (creator)
```
[the remark this answers is not in the document]
Comment (darshangodase) [reply backend/python/app/config/constants/arangodb.py]: fixed

File: backend/python/app/schema/arango/graph.py [hunk 4]
```diff
@@ -81,4 +81,9 @@
         "from_vertex_collections": [CollectionNames.USERS.value, CollectionNames.TEAMS.value, CollectionNames.ROLES.value, CollectionNames.GROUPS.value, CollectionNames.ORGS.value],
         "to_vertex_collections": [CollectionNames.AGENT_INSTANCES.value, CollectionNames.AGENT_TEMPLATES.value, CollectionNames.TEAMS.value, CollectionNames.ROLES.value, CollectionNames.RECORDS.value, CollectionNames.RECORD_GROUPS.value ],
     },
+    {
+        "edge_collection": CollectionNames.TICKET_RELATIONS.value,
+        "from_vertex_collections": [CollectionNames.TICKETS.value, CollectionNames.RECORDS.value],
+        "to_vertex_collections": [CollectionNames.USERS.value, CollectionNames.RECORDS.value, CollectionNames.TICKETS.value],
```
[the remark this answers is not in the document]
Comment (darshangodase) [reply backend/python/app/schema/arango/graph.py]: fixed

File: backend/python/app/config/constants/arangodb.py [hunk 5]
```diff
@@ -261,6 +262,35 @@ class RecordRelations(Enum):
     PARENT_CHILD = "PARENT_CHILD"
     SIBLING = "SIBLING"
     ATTACHMENT = "ATTACHMENT"
+    DUPLICATE = "DUPLICATE"
+    OTHER = "OTHERS"
```
[the remark this answers is not in the document]
Comment (darshangodase) [reply backend/python/app/config/constants/arangodb.py]: fixed

File: backend/python/app/connectors/utils/ticket_mapper.py [hunk 6]
```diff
@@ -0,0 +1,375 @@
...
+    if normalized in tag_mappings:
+        return tag_mappings[normalized]
+
+    # Try to find partial match (e.g., "relates to" matches "relates_to")
+    for key, value in tag_mappings.items():
+        if key.replace("_", " ") == normalized or normalized.replace(" ", "_") == key:
+            return value
+
+    # If no match found, return original value to preserve connector-specific tag
+    return api_tag
```
[the remark this answers is not in the document]
Comment (darshangodase) [reply backend/python/app/connectors/utils/ticket_mapper.py:375]: fixed
